thali/thalireplicationmanager.js

```diff
@@ -38,36 +39,70 @@ ThaliReplicationManager.events = {
 };
 
 /**
-* Starts the Thali replication manager with the given device name and port number
-* @param {String} deviceName the device name to advertise.
+* Starts the Thali replication manager with the given port number and db name
 * @param {Number} port the port number used for synchronization.
+* @param {String} dbName the db name to sync.
 */
-ThaliReplicationManager.prototype.start = function (deviceName, port, dbName) {
-  validations.ensureNonNullOrEmptyString(deviceName, 'deviceName');
+ThaliReplicationManager.prototype.start = function (port, dbName) {
   validations.ensureValidPort(port);
   validations.ensureNonNullOrEmptyString(dbName, 'dbName');
 
-  this.emit(ThaliReplicationManager.events.STARTING);
-
-  this._port = port;
-  this._deviceName = deviceName;
-  this._dbName = dbName;
-  this._serverBridge = muxServerBridge.call(this, port);
-  this._serverBridge.listen(function () {
-    this._serverBridgePort = this._serverBridge.address().port;
+  var cryptomanager = require('./thalicryptomanager');
+  // get the device public-key-hash
+  cryptomanager.getPublicKeyHash(function (err, publicKeyHash) {
+    var currentAddrEntry;
+    if (err) {
+      console.log('getPublicKeyHash returned err: ', err);
+      this._isStarted = false;
+      this.emit(ThaliReplicationManager.events.START_ERROR, err);
+      return;
+    } else {
+      console.log('got a valid publicKeyHash');
+      currentAddrEntry = addressPrefix + publicKeyHash;
+    }
 
-    this._emitter.startBroadcasting(deviceName, this._serverBridgePort, function (err) {
-      if (err) {
-        this._isStarted = false;
-        this.emit(ThaliReplicationManager.events.START_ERROR, err);
-      } else {
-        this._isStarted = true;
-        this._emitter.addListener(PEER_AVAILABILITY_CHANGED, syncPeers.bind(this));
-        this._emitter.addListener(NETWORK_CHANGED, networkChanged.bind(this));
-        this.emit(ThaliReplicationManager.events.STARTED);
+    // check if the db already has an addressbook entry for the public-key-hash
+    this._db.allDocs({
+      include_docs: true,
+      attachments: true
+    }).then(function (result) {
+      var currentAdddrFound = false;
```

Comment:
There should be 2, not 3 ds in Addr. Actually it's generally not recommended to use abbreviations. Now that we have auto-complete in our IDEs it's better to just use the full word, e.g. currentAddressFound.